File: frontend/src/app/model/dto/home-dto.ts
import { HomeMaterial } from "../enum/home-material";
import { FloorMaterial } from "../enum/floor-material";
import { NecessityDTO } from "./necessity-dto";
import { PersonDTO } from "./person-dto";

export class HomeDTO {
    id: number;
    complement: string;
    isRented: boolean;
    necessities: NecessityDTO[];
    persons: PersonDTO[];
    averageFamilyIncome: number;
    material: HomeMaterial;
    floorMaterial: FloorMaterial;
    hasWaterTank: boolean;
    hasWaterPipe: boolean;
    hasSepticTank: boolean;
    roomsNumber: number;
    bathroomsNumber: number;

    constructor(
        id: number,
        complement: string,
        isRented: boolean,
        necessities: NecessityDTO[],
        persons: PersonDTO[],
        averageFamilyIncome: number,
        material: HomeMaterial,
        floorMaterial: FloorMaterial,
        hasWaterTank: boolean,
        hasWaterPipe: boolean,
        hasSepticTank: boolean,
        roomsNumber: number,
        bathroomsNumber: number
    ) {
        this.id = id;
        this.complement = complement;
        this.isRented = isRented;
        this.necessities = necessities;
        this.persons = persons;
        this.averageFamilyIncome = averageFamilyIncome;
        this.material = material;
        this.floorMaterial = floorMaterial;
        this.hasWaterTank = hasWaterTank;
        this.hasWaterPipe = hasWaterPipe;
        this.hasSepticTank = hasSepticTank;
        this.roomsNumber = roomsNumber;
        this.bathroomsNumber = bathroomsNumber;
    }
}

File: backend/src/main/java/com/viniciusfinger/rsflood/dto/HomeDTO.java
package com.viniciusfinger.rsflood.dto;

import com.viniciusfinger.rsflood.enums.FloorMaterial;
import com.viniciusfinger.rsflood.enums.HomeMaterial;
import com.viniciusfinger.rsflood.model.Address;
import com.viniciusfinger.rsflood.model.HomeStructure;
import lombok.Data;

import java.util.List;

@Data
public class HomeDTO {

    private Long id;

    private Address address;

    // Apartment number, home number in yard...
    private String complement;

    private Boolean isRented;

    private List<NecessityDTO> necessities;

    private List<PersonDTO> persons;

    private Double averageFamilyIncome;

    private HomeMaterial material;

    private FloorMaterial floorMaterial;

    //Caixa d'água
    private Boolean hasWaterTank;

    //Água encanada
    private Boolean hasWaterPipe;

    //Fossa séptica
    private Boolean hasSepticTank;

    private Integer roomsNumber;

    private Integer bathroomsNumber;

}
